pkg/apis/eventsource/v1alpha1/types.go

```diff
@@ -461,6 +461,22 @@ type StorageGridEventSource struct {
 	Events []string `json:"events,omitempty" protobuf:"bytes,2,opt,name=events"`
 	// Filter on object key which caused the notification.
 	Filter *StorageGridFilter `json:"filter,omitempty" protobuf:"bytes,3,opt,name=filter"`
+	// TopicArn
+	TopicArn string `json:"topicArn" protobuf:"bytes,4,name=topicArn"`
+	// Name of the bucket to register notifications for.
+	Bucket string `json:"bucket" protobuf:"bytes,5,name=bucket"`
+	// S3 region.
+	// Defaults to us-east-1
+	// +optional
+	Region string `json:"region,omitempty" protobuf:"bytes,6,opt,name=region"`
+	// Auth token for storagegrid api
+	AuthToken *corev1.SecretKeySelector `json:"authToken" protobuf:"bytes,7,name=authToken"`
+	// ApiURL is the url of the storagegrid api.
+	ApiURL string `json:"apiURL" protobuf:"bytes,8,name=apiURL"`
+	// Namespace refers to Kubernetes namespace to read auth token from.
+	// Defaults to gateway namespace
+	// +optional
+	Namespace string `json:"namespace,omitempty" protobuf:"bytes,9,opt,name=namespace"`
```

Comment:
It doesn't make sense to read a secret from a different namespace. If this is allowed, it means the SA should have the privilege to access all the secrets in the cluster, if the app is hacked, all the secrets in the cluster are exposed.